### PR TITLE
feature #495 : clean summary file

### LINKED ISSUE #495: [Feature] summary file cleaning
**Is your feature request related to a problem? Please describe.**
The summary issue keeps appending, which can lead to oversized files and long recovery times。

**Describe the solution you'd like**
Summary files need to delete from disk, design and implement  a mechanism.
After the mechanism run, the file does not appear fragmented and oversized.

**Describe alternatives you've considered**
It is also possible to not use the operation, but with possible system downtime or reboot, the files will show fragmentation.

**Additional context**
N.A

### PR DESCRIPTION
**Pull request by Subsegment**
# Which issue does this PR close?

Closes #495 .

 # Rationale for this change

improve summary

# What changes are included in this PR?

when summary size exceed max_summary_size, create a new file and write summary state to it, then move to replace origin summary file

# Are there any user-facing changes?

### CODE REVIEW

File: tskv/src/summary.rs [hunk 1]
```diff
@@ -284,6 +294,69 @@ impl Summary {
...
+                                self.ctx.last_seq(),
+                                max_level_ts,
+                                meta,
+                            );
+                        }
+                    }
+                    edits.push(edit);
+                }
+            }
+            let new_path = &file_utils::make_summary_file(GLOBAL_CONFIG.db_path.clone(), 1);
```
Comment (roseboy-liu) [inline tskv/src/summary.rs:335]: summary.tmp

File: tskv/src/summary.rs [hunk 2]
```diff
@@ -284,6 +294,69 @@ impl Summary {
...
+            let old_path = &self.writer.path().clone();
+            if try_exists(new_path) {
+                match remove_file(new_path) {
+                    Ok(_) => (),
+                    Err(e) => {
+                        error!("failed remove file {:?}, in case {:?}", new_path, e);
+                    }
+                };
+            }
+            self.writer = Writer::new(new_path).unwrap();
```
Comment (roseboy-liu) [inline tskv/src/summary.rs:345]: reopen the write after  ‘rename’ action

**Review by zipper-meng — APPROVED** [review]
OK.

**Comment by zipper-meng** [comment]
Ok.

**Review by zipper-meng — APPROVED** [review]
Ok.

**Pull request merged**